sound/soc/sof/sof-client-dma-trace.c

```diff
@@ -348,11 +374,13 @@ static int sof_dfsentry_trace_release(struct inode *inode, struct file *file)
 	if (!priv->dtrace_is_enabled)
 		priv->host_offset = 0;
 
+	debugfs_file_put(file->f_path.dentry);
+	sof_client_core_put(cdev);
 	return 0;
 }
 
 static const struct file_operations sof_dtrace_trace_fops = {
-	.open = simple_open,
+	.open = sof_dfsentry_trace_open,
```

Comment:
That is also not clear to me, do we really need to protect. What happens if we allow the parent to be removed, the trace would just exit as well, no?

Reply:
Yes it will, but the debugfs releasae will be called most of the time if not always _after_ the module has been removed and the access to random memory is not well tolerated, which we do in the release of the debugfs file.